### PR TITLE
PAINTROID 443 Added main options to landing page 

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
Implemented an overflow menu for the landing page top bar and added the following options from the current state of Paintroid: Rate us, Help(the onboarding screen will be implemented in a separate ticket), About, Feedback.

Ticket https://jira.catrob.at/browse/PAINTROID-443

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
         android:maxSdkVersion="28" />
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" />
```
Comment (juliajulie95) [inline android/app/src/main/AndroidManifest.xml]: Please do not use this permission.
Here is a guide: https://developer.android.com/training/package-visibility
https://developer.android.com/training/package-visibility/use-cases

**Comment by juliajulie95** [comment]
There are static analyze errors, please have a look

File: lib/io/src/ui/about_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  static const urlLicense = 'https://developer.catrobat.org/licenses';
+  static const urlCatrobat = 'https://catrobat.org';
+
+  static const urlTextStyle = TextStyle(
+    color: Color(0xFFE68B00),
+    fontSize: 18,
+    decoration: TextDecoration.underline,
+  );
+
+  TextSpan _clickableText(String text, String url, TextStyle? style) =>
```
Comment (juliajulie95) [inline lib/io/src/ui/about_dialog.dart:39]: Again function that returns a widget.
Use a class

File: lib/ui/main_overflow_menu.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        break;
+      case MainOverflowMenuOption.help:
+        break;
+      case MainOverflowMenuOption.about:
+        if (mounted) {
+          showMyAboutDialog(context, version);
+        }
+        break;
+      case MainOverflowMenuOption.feedback:
+        final url = Uri.parse(feedbackUrl);
```
Comment (juliajulie95) [inline lib/ui/main_overflow_menu.dart]: We could make a util that has this as a function and reuse it from there. Because I already saw it somewhere else

**Review by juliajulie95 — CHANGES_REQUESTED** [review]

File: lib/io/src/ui/about_dialog.dart (hunk 2 above)
Comment (CodeChamp-SS) [reply lib/io/src/ui/about_dialog.dart:39]: I also thought of using a class for this but actually, `TextSpan` is not a widget 😅

Comment (juliajulie95) [reply lib/io/src/ui/about_dialog.dart:39]: True, didn't know that :)

**Review by juliajulie95 — APPROVED** [review]

File: lib/io/src/ui/about_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:paintroid/io/src/ui/delete_project_dialog.dart';
+
+import '../../../ui/color_schemes.dart';
+import '../../../ui/util.dart';
```
Comment (ThorstenBandel) [inline lib/io/src/ui/about_dialog.dart]: I am pretty sure it should be able to not be imported being dependent on the path or?

Comment (CodeChamp-SS) [reply lib/io/src/ui/about_dialog.dart]: ah.. do you mean like these imports can be removed or?

Comment (ThorstenBandel) [reply lib/io/src/ui/about_dialog.dart]: Not removed but changed. The project is currently only using package imports and we should stay consistent :D 
That's why i would say we change those to package imports as well. I just saw that there is one more of those imports in the file landing_page.dart

Comment (CodeChamp-SS) [reply lib/io/src/ui/about_dialog.dart]: ah.. alright. Will just fix it.

**Pull request merged**